prospero-common/src/main/java/org/wildfly/prospero/galleon/ChannelMavenArtifactRepositoryManager.java

```diff
@@ -59,70 +57,83 @@ public void resolve(MavenArtifact artifact) throws MavenUniverseException {
                 result = channelSession.resolveMavenArtifact(artifact.getGroupId(), artifact.getArtifactId(), artifact.getExtension(),
                         artifact.getClassifier(), null);
             } else {
-                final DefaultArtifact gav = new DefaultArtifact(artifact.getGroupId(), artifact.getArtifactId(), artifact.getClassifier(), artifact.getExtension(), artifact.getVersion() != null ? artifact.getVersion() : artifact.getVersionRange());
-
-                Optional<DefaultArtifact> found = manifest.findStreamFor(gav.getGroupId(), gav.getArtifactId()).map(this::streamToArtifact);
+                Optional<DefaultArtifact> found = manifest.findStreamFor(artifact.getGroupId(), artifact.getArtifactId()).map(this::streamToArtifact);
 
                 if (found.isPresent()) {
                     result = channelSession.resolveDirectMavenArtifact(artifact.getGroupId(), artifact.getArtifactId(), artifact.getExtension(),
                             artifact.getClassifier(), found.get().getVersion());
-                } else if (artifact.getArtifactId().equals("community-universe") || artifact.getArtifactId().equals("wildfly-producers")) {
+                } else if (isUniverseOrProducerArtifact(artifact.getArtifactId())) {
                     result = channelSession.resolveMavenArtifact(artifact.getGroupId(), artifact.getArtifactId(), artifact.getExtension(),
                             artifact.getClassifier(), null);
                 } else {
                     throw new MavenUniverseException("Unable to resolve " + artifact);
                 }
             }
-            artifact.setVersion(result.getVersion());
-            artifact.setPath(result.getFile().toPath());
+
+            MavenArtifactMapper.resolve(artifact, result);
         } catch (UnresolvedMavenArtifactException e) {
             throw new MavenUniverseException(e.getLocalizedMessage(), e);
         }
     }
 
-    private DefaultArtifact streamToArtifact(Stream s) {
-        return new DefaultArtifact(s.getGroupId(), s.getArtifactId(), "jar", s.getVersion());
-    }
+    @Override
+    public void resolveAll(Collection<MavenArtifact> artifacts) throws MavenUniverseException {
```

Comment:
I'm asking to be sure - do I assume correctly that this method should be functionally equivalent to the singular resolve() above?

Reply:
Yes. The outcome of calling resolve on a set of artifacts should be the same as calling resolveAll on that set